enumerable_method.rb

```diff
@@ -0,0 +1,145 @@
+# frozen_string_literal: true
+
+module Enumerable #:nodoc:
+  # my_each method
+
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(to_a[i])
+      i += 1
+    end
+    self
+  end
+
+  # each_with_index method
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i <= length - 1
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  # my_select method
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    select = []
+    if block_given?
+      my_each do |x|
+        element = yield(x)
+        select << x if element == true
+      end
+    end
+    select
+  end
+
+  # my_all method
+
+  def my_all?(arg = nil)
+    arr = self
+    if block_given? && arg.nil?
+      arr.my_each { |item| return false unless item == true }
+    elsif arg
+      arr.my_each { |item| return false unless clases(item, arg) }
+    elsif !block_given? && arg.nil?
+      arr.my_each { |item| return false unless item }
+    end
+    true
+  end
+
+  # my_any? method
+
+  def my_any?(arg = nil)
+    arr = self
+    if !block_given? && arg.nil?
+      arr.my_each { |item| return true if item }
+    elsif !block_given? && arg
+      arr.my_each { |item| return true if clases(item, arg) }
+    end
+    false
+  end
+
+  # my_none? method
+
+  def my_none?(arg = nil)
+    arr = self
+    if !block_given? && arg.nil?
+      arr.my_each { |item| return false if item }
+    elsif !block_given? && arg
+      arr.my_each { |item| return false if clases(item, arg) }
+    end
+    true
+  end
```

Comment:
**[REQUIRED]**
* Returns true if the block never returns true for all elements.
```
Failure/Error: expect(array.my_none?(&false_block)).to eq(array.none?(&false_block))
     
       expected: false
            got: true
     
       (compared using ==)
```